src/artemangaweb/contexto_comun.py

```diff
@@ -6,7 +6,7 @@
 from django.db.models import Q
 
 def obtener_editoriales_y_categorias(request):
-    editoriales = Editorial.objects.all()
+    editoriales = Editorial.objects.all().filter(producto__stock__gte=1).filter(producto__esta_publicado=True).distinct()[:8]
```

Comment:
qué hace ``distinct()``?

Reply:
puedes omitir el ``all()`` y remplazarlo directamente con el ``filter()``

Reply:
El distinct lo  cambie un por filtro del tipo "and" en la query, queda más limpio.